### PR TITLE
tex-fmt: init at 0.1.0

### PR DESCRIPTION
**Pull request by WGUNDERWOOD**
## Description of changes

tex-fmt is a LaTeX formatter written in Rust.
https://github.com/WGUNDERWOOD/tex-fmt/
It is a work in progress but the current version v0.1.0 has basic functionality.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by yunfachi — CHANGES_REQUESTED** [review]
Rename the pull request to `tex-fmt: init at 0.1.0` and leave two commits with these messages:
```
maintainers: add wgunderwood
tex-fmt: init at 0.1.0
```

File: pkgs/by-name/te/tex-fmt/package.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "tex-fmt";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "wgunderwood";
+    repo = "tex-fmt";
+    rev = "v${version}";
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/by-name/te/tex-fmt/package.nix:10]: > Update scripts sometimes get confused by this

Is this needed? Usually we only require this when the upstream uses a confusing tag system.

Further, it should be considered a bug on the update scripts.

[the remark this answers is not in the document]
Comment (yunfachi) [reply pkgs/by-name/te/tex-fmt/package.nix:10]: Сonfused the repository with another one, sorry

File: pkgs/by-name/te/tex-fmt/package.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    hash = "sha256-FfBJV7zIdGZvEOlM37DGbbs4dvIlwmTWTenWn434dIY=";
+  };
+
+  cargoHash = "sha256-4Vgzg7IpftUtqJK91qiW+pwe2SrMHtk4EgsAlk3f3Pk=";
+
+  meta = with lib; {
+    description = "A LaTeX formatter written in Rust.";
```
[the remark this answers is not in the document]
Comment (WGUNDERWOOD) [reply pkgs/by-name/te/tex-fmt/package.nix]: Fixed in new commit

File: pkgs/by-name/te/tex-fmt/package.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    homepage = "https://github.com/wgunderwood/tex-fmt";
+    license = licenses.mit;
+    mainProgram = "tex-fmt";
+    maintainers = with maintainers; [ wgunderwood ];
+    platforms = platforms.linux;
```
[the remark this answers is not in the document]
Comment (WGUNDERWOOD) [reply pkgs/by-name/te/tex-fmt/package.nix]: Fixed in new commit.

File: pkgs/by-name/te/tex-fmt/package.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "wgunderwood";
+    repo = "tex-fmt";
+    rev = "v${version}";
+    hash = "sha256-FfBJV7zIdGZvEOlM37DGbbs4dvIlwmTWTenWn434dIY=";
+  };
+
+  cargoHash = "sha256-4Vgzg7IpftUtqJK91qiW+pwe2SrMHtk4EgsAlk3f3Pk=";
+
+  meta = with lib; {
```
[the remark this answers is not in the document]
Comment (WGUNDERWOOD) [reply pkgs/by-name/te/tex-fmt/package.nix]: Fixed in new commit.

File: pkgs/by-name/te/tex-fmt/package.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (WGUNDERWOOD) [reply pkgs/by-name/te/tex-fmt/package.nix]: Fixed in new commit.

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -21498,6 +21498,12 @@
     githubId = 24979302;
     name = "Vladimír Zahradník";
   };
+  wgunderwood = {
+    email = "wg.underwood13@gmail.com";
+    github = "wgunderwood";
```
[the remark this answers is not in the document]
Comment (WGUNDERWOOD) [reply maintainers/maintainer-list.nix]: Fixed in new commit.

**Comment by yunfachi** [comment]
LGTM, but the `tex-fmt: init at 0.1.0` commit should be after `maintainers: add wgunderwood`.

**Comment by WGUNDERWOOD** [comment]
> LGTM, but the `tex-fmt: init at 0.1.0` commit should be after `maintainers: add wgunderwood`.

I have switched the order of these commits.

**Review by yunfachi — APPROVED** [review]

File: pkgs/by-name/te/tex-fmt/package.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  cargoHash = "sha256-4Vgzg7IpftUtqJK91qiW+pwe2SrMHtk4EgsAlk3f3Pk=";
+
+  meta = {
+    description = "A LaTeX formatter written in Rust";
+    homepage = "https://github.com/WGUNDERWOOD/tex-fmt";
+    license = lib.licenses.mit;
+    mainProgram = "tex-fmt";
+    maintainers = with lib.maintainers; [ wgunderwood ];
+    platforms = lib.platforms.linux;
```
Comment (AndersonTorres) [inline pkgs/by-name/te/tex-fmt/package.nix:22]:
```suggestion
```

This is auto-set by `rustPlatform.buildRustPackage`; you should set this only when the upstream explicitly says something like  "Linux-only".

Comment (WGUNDERWOOD) [reply pkgs/by-name/te/tex-fmt/package.nix]: Thanks, I have fixed this.

**Comment by AndersonTorres** [comment]
The code is fine.

But, instead of using `git merge` to align this PR with the master branch, use `git rebase`.

**Pull request closed without merging**